test/page-network-idle.spec.ts

```diff
@@ -16,123 +16,106 @@
-import type { Frame, Page } from '..';
-import { TestServer } from '../utils/testserver';
 
 it('should navigate to empty page with networkidle', async ({page, server}) => {
   const response = await page.goto(server.EMPTY_PAGE, { waitUntil: 'networkidle' });
   expect(response.status()).toBe(200);
 });
 
-async function networkIdleTest(frame: Frame, server: TestServer, action: () => Promise<any>, isSetContent?: boolean) {
-  const finishResponse = response => {
-    response.statusCode = 404;
-    response.end(`File not found`);
-  };
-  const waitForRequest = suffix => {
-    return Promise.all([
-      server.waitForRequest(suffix),
-      (frame['_page'] as Page).waitForRequest(server.PREFIX + suffix),
-    ]);
-  };
-  const responses = {};
-  // Hold on to a bunch of requests without answering.
-  server.setRoute('/fetch-request-a.js', (req, res) => responses['a'] = res);
-  const firstFetchResourceRequested = waitForRequest('/fetch-request-a.js');
-  server.setRoute('/fetch-request-d.js', (req, res) => responses['d'] = res);
-  const secondFetchResourceRequested = waitForRequest('/fetch-request-d.js');
-
-  const waitForLoadPromise = isSetContent ? Promise.resolve() : frame.waitForNavigation({ waitUntil: 'load' });
-
-  // Navigate to a page which loads immediately and then does a bunch of
-  // requests via javascript's fetch method.
-  const actionPromise = action();
-
-  // Track when the action gets completed.
-  let actionFinished = false;
-  actionPromise.then(() => actionFinished = true);
-
-  // Wait for the frame's 'load' event.
-  await waitForLoadPromise;
-  expect(actionFinished).toBe(false);
-
-  // Wait for the initial resource to be requested.
-  await firstFetchResourceRequested;
-  expect(actionFinished).toBe(false);
-
-  expect(responses['a']).toBeTruthy();
-  // Finishing response should trigger the second round.
-  finishResponse(responses['a']);
-
-  // Wait for the second round to be requested.
-  await secondFetchResourceRequested;
-  expect(actionFinished).toBe(false);
-  // Finishing the last response should trigger networkidle.
-  let timerTriggered = false;
-  const timer = setTimeout(() => timerTriggered = true, 500);
-  finishResponse(responses['d']);
-
-  const response = await actionPromise;
-  clearTimeout(timer);
-  expect(timerTriggered).toBe(true);
-  if (!isSetContent)
-    expect(response.ok()).toBe(true);
-}
-
-it('should wait for networkidle to succeed navigation', async ({page, server}) => {
-  await networkIdleTest(page.mainFrame(), server, () => {
-    return page.goto(server.PREFIX + '/networkidle.html', { waitUntil: 'networkidle' });
+function networkIdleTest(method: 'goto' | 'gotoIframe' | 'waitForNavigation' | 'setContent' | 'setContentIframe', childFrame: boolean, staleRequest: boolean) {
+  const title = `should wait for networkidle in ${childFrame ? 'child' : 'main'} frame during ${method}${staleRequest ? 'with stale request' : ''}`;
+
+  it(title, async ({page, server}) => {
```

Comment:
let's make sure tests have a long common prefix inlined next to `it`
`it('should wait for networkidle in _frame_ during _method_ with _stale_ request: [frame=foo, method=goto, stale=true]`